pkg/controllers/provisioning/controller.go

```diff
@@ -101,9 +101,8 @@ func (c *Controller) Apply(ctx context.Context, provisioner *v1alpha5.Provisione
 		return err
 	}
 	provisioner.Spec.Labels = functional.UnionStringMaps(provisioner.Spec.Labels, map[string]string{v1alpha5.ProvisionerNameLabelKey: provisioner.Name})
-	provisioner.Spec.Requirements = provisioner.Spec.Requirements.
-		Add(cloudprovider.Requirements(instanceTypes).Requirements...).
-		Add(v1alpha5.NewLabelRequirements(provisioner.Spec.Labels).Requirements...)
+	provisioner.Spec.Requirements.Combine(cloudprovider.Requirements(instanceTypes).Requirements...)
+	provisioner.Spec.Requirements.Combine(v1alpha5.NewLabelRequirements(provisioner.Spec.Labels).Requirements...)
```

Comment:
If we're adding all labels into requirements, would it be possible to not write the labels into `provisioner.spec.labels` and threat that as an api-only syntactic sugar field?